repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py

```diff
@@ -0,0 +1,213 @@
+            if sp == req_sp:
+                rev = int(number.lstrip('0') or '0')
+                if rev < req_rev:
+                    continue
+                if rev == req_rev:
+                    patch_level = int(_manifest_get(instance.manifest, 'rev-patchlevel', '00').lstrip('0') or '0')
+                    if patch_level < req_pl:
+                        continue
+                return True
+        return False
+    # if not 'len(number) > 2 and number.isdigit()'
+    api.current_logger().warn(
+        'Invalid rev-number field value `{}` in manifest for instance {}'.format(number, instance.name))
+    return False
+
+
+def _fullfills_hana_min_version(instance):
+    """ Performs a check whether the version of SAP HANA fullfills the minimal requirements for the target RHEL """
+    return _major_version_check(instance) and _sp_rev_patchlevel_check(instance)
+
+
+def version2_check(info):
+    """ Performs all checks for SAP HANA 2 and creates a report if anything unsupported has been detected """
+    found = {}
+    for instance in info.instances:
+        if _manifest_get(instance.manifest, 'release', None) == '1.00':
+            continue
+        if not _fullfills_hana_min_version(instance):
+            _add_hana_details(found, instance)
+
+    if found:
+        detected = _create_detected_instances_list(found)
+        reporting.create_report([
+            reporting.Title('SAP HANA needs to be updated before upgrade'),
+            reporting.Summary(
+                ('A newer version of SAP HANA is required in order continue with the upgrade.'
+                 ' {min_hana_version} is required for the target version of RHEL.\n\n'
+                 'The following SAP HANA instances have been detected to be running with a lower version'
+                 ' than required on the target system:\n'
+                 '{detected}').format(detected=detected, min_hana_version=SAP_HANA_MINIMAL_VERSION_STRING)
+            ),
+            reporting.RemediationHint('Update SAP HANA at least to {}'.format(SAP_HANA_MINIMAL_VERSION_STRING)),
+            reporting.ExternalLink(url='https://launchpad.support.sap.com/#/notes/2235581',
+                                   title='SAP HANA: Supported Operating Systems'),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SANITY]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Audience('sysadmin')
+        ])
+
+
+def platform_check():
+    """ Creates an inhibitor report in case the system is not running on x86_64 """
+    if not architecture.matches_architecture(architecture.ARCH_X86_64):
+        reporting.create_report([
+            reporting.Title('SAP HANA upgrades are only supported on X86_64 systems'),
+            reporting.Summary(
+                ('Upgrades for SAP HANA are only supported on X86_64 systems.'
+                 ' For more information please consult the documentation.')
```

Comment:
The supported hardware architectures are also contained in https://launchpad.support.sap.com/#/notes/2235581.